python/flashinfer/decode.py

```diff
@@ -78,42 +79,173 @@ def compile_batch_decode_module(
-                run=run_func,
-            )
         else:
-            _batch_decode_modules[args] = compile_batch_decode_module(*args)
+            mod = compile_batch_decode_module(*args)
+            plan_func = mod.plan
+            run_func = mod.run
+
+        # torch library for batch_decode_with_paged_kv_cache_run
+
+        @register_custom_op(
+            f"flashinfer::{uri}_run",
+            mutates_args=(
+                "float_workspace_buffer",
+                "int_workspace_buffer",
+                "paged_k_cache",
+                "paged_v_cache",
+                "maybe_lse",
+            ),
+        )
+        def run_batch_decode(
+            float_workspace_buffer: torch.Tensor,
+            int_workspace_buffer: torch.Tensor,
+            plan_info_vec: List[int],
+            q: torch.Tensor,
+            paged_k_cache: Optional[torch.Tensor],
+            paged_v_cache: Optional[torch.Tensor],
+            paged_kv_indptr: torch.Tensor,
+            paged_kv_indices: torch.Tensor,
+            paged_kv_last_page_len: torch.Tensor,
+            alibi_slopes: Optional[torch.Tensor],
+            kv_layout_code: int,
+            window_left: int,
+            logits_soft_cap: float,
+            sm_scale: float,
+            rope_scale: float,
+            rope_theta: float,
+            maybe_lse: Optional[torch.Tensor],
+        ) -> torch.Tensor:
+            return run_func(
+                float_workspace_buffer,
+                int_workspace_buffer,
+                plan_info_vec,
+                q,
+                paged_k_cache,
+                paged_v_cache,
+                paged_kv_indptr,
+                paged_kv_indices,
+                paged_kv_last_page_len,
+                alibi_slopes,
+                kv_layout_code,
+                window_left,
+                logits_soft_cap,
+                sm_scale,
+                rope_scale,
+                rope_theta,
+                maybe_lse,
+            )
+
+        @register_fake_op(f"flashinfer::{uri}_run")
+        def _fake_run_batch_decode(
+            float_workspace_buffer: torch.Tensor,
+            int_workspace_buffer: torch.Tensor,
+            plan_info_vec: List[int],
+            q: torch.Tensor,
+            paged_k_cache: Optional[torch.Tensor],
+            paged_v_cache: Optional[torch.Tensor],
+            paged_kv_indptr: torch.Tensor,
+            paged_kv_indices: torch.Tensor,
+            paged_kv_last_page_len: torch.Tensor,
+            alibi_slopes: Optional[torch.Tensor],
+            kv_layout_code: int,
+            window_left: int,
+            logits_soft_cap: float,
+            sm_scale: float,
+            rope_scale: float,
+            rope_theta: float,
+            maybe_lse: Optional[torch.Tensor],
+        ) -> torch.Tensor:
+            return torch.empty_like(q)
+
+        # Register the module.
+        #
+        # Note that plan is not part of model logic. It should not be included in
```

Comment:
Yes and we should note it in the documentation.

Reply:
Added